examples/gtest_emulation/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2020  ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.13)
+
+project(gtest CXX)
+
+if(EXISTS "${HPX_DIR}")
+
+  find_package(HPX REQUIRED)
+
+  # Add a static library which contains a main to emulate gtest_main
+  add_library(static_main_lib STATIC static_main.cpp)
+
+  # /!\ This helper interface is needed to keep the right linking order
+  add_library(hpx_helper_interface INTERFACE)
+  target_link_libraries(
+    hpx_helper_interface INTERFACE HPX::hpx HPX::wrap_main static_main_lib
+  )
+
+  # Test with the main function in a separate static library
+  add_executable(hpx_main_ext_main hpx_main_ext_main.cpp)
+  target_link_libraries(hpx_main_ext_main PRIVATE hpx_helper_interface)
```

Comment:
This one would only work if `HPX_WITH_DYNAMIC_MAIN=ON`. I think the build tests don't get built on Windows GitHub actions... would you mind adding explicit commands to build the non-pkgconfig build tests to the Windows GitHub actions configs (there are two configurations) since pseudotargets don't work on Windows?

Reply:
will do